tests/units/hindawi/test_hindawi_text_extractor.py

```diff
@@ -0,0 +1,56 @@
+import xml.etree.ElementTree as ET
+
+from common.parsing.xml_extractors import RequiredFieldNotFoundExtractionError
+from hindawi.xml_extractors import HindawiTextExtractor
+from pytest import fixture, raises
+
+
+@fixture(scope="module")
+def hindawi_doi_extractor():
+    prefixes = {
+        "marc": "http://www.loc.gov/MARC21/slim",
+        "ns0": "http://www.openarchives.org/OAI/2.0/",
+        "ns1": "http://www.loc.gov/MARC21/slim",
+        "xsi": "http://www.w3.org/2001/XMLSchema-instance",
+    }
+    return HindawiTextExtractor(
+        source="ns0:metadata/ns1:record/ns0:datafield/[@tag='024']/ns0:subfield/[@code='a']",
+        destination="dois",
+        prefixes=prefixes,
+    )
+
+
+@fixture()
+def articles(shared_datadir):
+    articles = []
+    files = ["example1.xml", "example2.xml"]
```

Comment:
do they have the case that we want to test?

Reply:
@ErnestaP ☝️

Reply:
yes, the titles and abstracts have a lot of inner html's, like math expressions, italics

Reply:
@ErnestaP  hmm why it didn't fail before in the tests?

Reply:
because I didn't parse with mathlib at all, we left it for solving in the future.
The goal before was to extract the first string, to check are we parsing the right part in xml

Reply:
and also it was different parsing function,which was taking just the first string, but not the whole value